src/components/OneTimeCodeInput/index.tsx

```diff
@@ -0,0 +1,148 @@
+
+interface OneTimeCodeContainerProps {
+  length: number
+}
+
+interface OneTimeCodeInputProps {
+  length: number
+  id: string
+  hideLabel?: boolean
+  ariaLabel: string
+  label?: string
+  disabled?: boolean
+  helpText?: string
+  onChange?: (value: any) => void
+}
+
+const OneTimeCodeContainer = styled.section<OneTimeCodeContainerProps>`
+  display: grid;
+  grid-template-columns: repeat(${props => props.length}, 1fr);
+  grid-gap: ${space[8]};
+  input {
+    text-align: center;
+  }
+`
+
+const Input = styled.input`
+  ${fieldStyles};
+`
+
+export const OneTimeCodeInput: React.FC<OneTimeCodeInputProps> = ({
+  length = 5,
+  id,
+  hideLabel,
+  ariaLabel,
+  label,
+  disabled = false,
+  helpText,
+  onChange = () => {},
+}) => {
+  const [inputValues, setInputValues] = useState(new Array(length).fill(''))
+  const containerRef = useRef<HTMLInputElement>(null)
+
+  const handleChange = (
+    event: React.ChangeEvent<HTMLInputElement>,
+    position: number
+  ) => {
+    if (!containerRef.current) return
+
+    const inputs = containerRef.current.querySelectorAll('input')
+    const newInputValues = [...inputValues]
+
+    if (event.target.value.length === 0) return
+
+    if (event.target.value.length === 1) {
+      newInputValues[position] = event.target.value
+
+      const index = newInputValues.findIndex(value => value === '')
+      if (index !== -1) {
+        inputs[index].focus()
+      }
+      onChange(newInputValues.join(''))
+      return setInputValues(newInputValues)
+    }
+
+    if (event.target.value.length === length) {
+      const multipleValues = event.target.value.split('')
+      setInputValues(multipleValues)
+
+      onChange(multipleValues.join(''))
+      inputs[length - 1].focus()
+    }
+  }
+
+  const handleKeyUp = (event: React.KeyboardEvent, position: number) => {
+    if (!containerRef.current) return
+
+    const inputs = containerRef.current.querySelectorAll('input')
+    const newInputValues = [...inputValues]
+    const moveRight =
+      event.key === 'ArrowRight' && position !== inputValues.length - 1
+    const moveLeft = event.key === 'ArrowLeft' && position !== 0
+    const Backspace = event.key === 'Backspace'
+
+    if (inputValues[0] === '') return
+
+    if (moveRight) {
+      inputs[position + 1].focus()
+    }
+
+    if (moveLeft) {
+      inputs[position - 1].focus()
+    }
+
+    if (Backspace) {
+      if (inputs[position].value) {
+        newInputValues[position] = ''
+        onChange(newInputValues.join(''))
+        setInputValues(newInputValues)
+      } else {
+        newInputValues[position - 1] = ''
+        onChange(newInputValues.join(''))
+        setInputValues(newInputValues)
+        inputs[position - 1].focus()
+      }
+    }
+  }
+
+  return (
+    <>
+      {hideLabel ? (
+        <VisuallyHidden>
+          <LabelText>{label}</LabelText>
+        </VisuallyHidden>
+      ) : (
+        <LabelText>{label}</LabelText>
+      )}
```

Comment:
Currently the label is not actually a label, is there a reason for that?

Reply:
No, will fix this